### PR TITLE
[RS][DDCE-1288] Updated to sbt 1.3.13.

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-1288

In order to update to Scala 2.12 and sbt 1.3.13, it was necessary to move off bee-client and onto the hmrc libraries for http requests. The vast majority of this change is necessitated by that.

## Checklist PR Raiser
- [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
- [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
- [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
- [x]  I've squashed my commits - including the JIRA issue number in the commit message
- [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by RafSobol** [comment]
https://github.com/hmrc/service-manager-config/pull/3213

**Comment by RafSobol** [comment]
https://github.com/hmrc/build-jobs/pull/7566

File: conf/application.conf [hunk 1]
```diff
@@ -53,7 +54,7 @@ play.http.errorHandler = "uk.gov.hmrc.play.bootstrap.http.JsonErrorHandler"
 # ~~~~~
 # The secret key is used to secure cryptographics functions.
 # If you deploy your application to several instances be sure to use the same key!
-application.secret="OhMSYIWoItbVV9bvKjsJgUBv3HahBDK5SV1Ci4weOPT3m6K5QtwLxbHRplUa5y7S"
+play.http.secret.key="OhMSYIWoItbVV9bvKjsJgUBv3HahBDK5SV1Ci4weOPT3m6K5QtwLxbHRplUa5y7S"
```
[the remark this answers is not in the document]
Comment (RafSobol) [reply conf/application.conf:57]: as per the discussion: let's keep it here for now, until a time when we get around to getting a new key

File: app/uk/gov/hmrc/brm/connectors/ResponseParser.scala [hunk 2]
```diff
@@ -16,30 +16,38 @@
...
 
       BirthSuccessResponse(json)
-    } catch {
-      case e: Exception =>
-        BrmLogger.error(CLASS_NAME, "parse", s"unable to parse json: $e")
+    } match {
+      case Success(successResponse) => successResponse
+      case Failure(exception) =>
+        BrmLogger.error(CLASS_NAME, "parse", s"unable to parse json: $exception")
         // from 200 t0 500 or override somehow?
```
[the remark this answers is not in the document]
Comment (RafSobol) [reply app/uk/gov/hmrc/brm/connectors/ResponseParser.scala]: removed

File: test/uk/gov/hmrc/brm/connectors/AuthenticatorSpec.scala [hunk 3]
```diff
@@ -17,17 +17,47 @@
...
 
+  val mockHttpClient: DefaultHttpClient = mock[DefaultHttpClient]
+
   val testAuthenticator =
-    new Authenticator(testProxyConfig, testGroConfig, mock[HttpClientFactory], mock[ProxyAuthenticator], mock[CertificateStatus])
+    new Authenticator(testGroConfig, mock[ProxyAuthenticator], mock[CertificateStatus], mockHttpClient)
+
+  val mockResponseHandler: ResponseHandler = mock[ResponseHandler]
+  val mockErrorHandler: ErrorHandler = mock[ErrorHandler]
+  implicit val hc: HeaderCarrier = HeaderCarrier()
```
[the remark this answers is not in the document]
Comment (RafSobol) [reply test/uk/gov/hmrc/brm/connectors/AuthenticatorSpec.scala:43]: you don't need to mock headercarriers

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+lazy val compile = Seq(
+  "uk.gov.hmrc" %% "bootstrap-play-26" % "2.3.0",
+  "uk.gov.hmrc" %% "domain" % "5.10.0-play-26",
+  compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
+  "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
+)
+
+lazy val scope: String = "test"
```
[the remark this answers is not in the document]
Comment (RafSobol) [reply build.sbt:61]: moved it into the test dependency list

**Review by nathan-rees — APPROVED** [review]

**Pull request merged**